wgsl/index.bs

```diff
@@ -9597,29 +9914,53 @@ value|non-uniform=] during the execution of the function call.
+    * For each formal parameter that is a pointer in the [=address spaces/function=] address space,
         create a <dfn noexport>Value_return_i</dfn> node.
+        This represents the function's effect on the uniformity of the value pointed to by the *i*<sup>th</sup> parameter.
 * Walk over the syntax of the function, adding nodes and edges to the graph following the rules of the next sections ([[#func-var-value-analysis]], [[#uniformity-statements]], [[#uniformity-function-calls]], [[#uniformity-expressions]]), using [=CF_start=] as the starting control-flow for the function's body.
-* For each [=Value_return_i=] node, record which [=param_i=] nodes are reachable from it.
-* Look at which nodes are reachable from [=RequiredToBeUniform=].
-    * If this set includes the node [=MayBeNonUniform=], then reject the program.
-    * If this set includes [=CF_start=], then the [=call site tag=] for the function is [=CallSiteRequiredToBeUniform=].
-    * Otherwise, the [=call site tag=] is [=CallSiteNoRestriction=].
-    * For each [=param_i=] in this set, the corresponding [=parameter tag=] is [=ParameterRequiredToBeUniform=].
-    * Remove from the graph all nodes that have been visited.
-* If [=Value_return=] exists, look at which nodes are reachable from it
-    * If this set includes [=MayBeNonUniform=], then the [=function tag=] is [=ReturnValueMayBeNonUniform=].
-    * For each [=param_i=] in this set, the corresponding [=parameter tag=] is [=ParameterRequiredToBeUniformForReturnValue=].
-* For each [=Value_return_i=] node, look at which nodes are reachable from it
-    * If this set includes [=MayBeNonUniform=], the corresponding [=pointer parameter tag=] is [=PointerParameterMayBeNonUniform=].
-    * Otherwise, the corresponding [=pointer parameter tag=] is [=PointerParameterNoRestriction=].
-* If the [=function tag=] has not been assigned, then it is [=NoRestriction=].
-* For each parameter, if it has not been assigned a [=parameter tag=], then it is [=ParameterNoRestriction=].
-
-Note: The entire graph can be destroyed at this point. The tags listed above are all that we need to remember to analyze callers of this function.
+* Initialize as follows:
+    * The [=function tag=] is initialized to [=NoRestriction=].
+    * The [=call site tag=] is initialized to [=CallSiteNoRestriction=].
+    * The [=parameter tag=] for each [=param_i=] is initialized to [=ParameterNoRestriction=].
+    * The [=parameter return tag=] for each [=param_i=] is initialized to [=ParameterReturnNoRestriction=].
+    * The [=pointer parameter tag=] for each [=param_i=], if it exists, is initialized to [=PointerParameterNoRestriction=].
+* For each severity *S* in the order {[=severity/error=], [=severity/warning=], [=severity/info=]}, perform the following:
+    * Let *R.S* be the set of unvisited nodes reachable from [=RequiredToBeUniform.S=].
+    * Mark the nodes in *R.S* as visited.
```

Comment:
> Mark the [=interior nodes=] in *R.S* as visited.

The reason we want this is so that we don't think of things like param_i as having been visited.  We still need to be able to "reach" them with a search via unvisited nodes.